xwiki-platform-core/xwiki-platform-xar/xwiki-platform-xar-model/src/main/java/org/xwiki/xar/XarPackage.java

```diff
@@ -231,6 +237,35 @@ public void read(ZipFile zipFile) throws IOException, XarException
         }
     }
 
+    /**
+     * Returns an input stream corresponding to the given localDocumentReference from the current XarPackage.
+     * This method throws an IOException if the entry or the file cannot be found.
+     *
+     * If the current XAR is packaged in a ZipFile, then the InputStream is directly retrieved from
+     * {@link XarFile#getInputStream(LocalDocumentReference)}, else a FileInputStream is created from the file
+     * corresponding to the document reference if it exists.
+     *
+     * @param localDocumentReference the reference for which to return the InputStream.
+     * @return an InputStream corresponding to the given reference.
+     * @throws IOException in case the entry or the file cannot be found.
+     * @throws IOException in case of errors when creating the {@link XarFile}.
+     * @since 10.11RC1
+     */
+    public InputStream getInputStream(LocalDocumentReference localDocumentReference) throws IOException, XarException
```

Comment:
I really don't like having this kind of API in XarPackage, it's not supposed to be its job.

If you really need a XarFile equivalent for folders then you should introduce one IMO.

Reply:
> If you really need a XarFile equivalent for folders then you should introduce one IMO.

Actually it's linked to the process of xwiki-platform-distribution-flavor-test-webstandards which I don't fully understand: during the preparation before the test, pages are exported in a folder, but without packaging it into a XAR, and then the `XarPackage.getEntries()` is used to read them.

So maybe we should just package the XAR?